### DISCUSSION TITLE
Add ability to group skin elements into a cluster

### DISCUSSION THREAD

**Discussion by nagi-desuuu** [ppy/osu#22911] [Ideas]
Selecting multiple skin elements and dragging them together is one thing, but doing this repetitively would be another thing. An element grouping feature would be handy, as it makes the interface treat a group of elements into a single cluster of elements.
(Highlighted section could be potential location of the group feature)

![image](https://user-images.githubusercontent.com/75596775/226180388-975b080f-80a1-4767-a11d-3fd166ae15d5.png)

If possible, a sub-feature could also be implemented so that, with an additional click of an element inside that cluster, the game selects that skin element attached to that group.
(Might need to watch out for https://github.com/ppy/osu/pull/22804 if it would be implemented)
I've used Adobe Express as an example of how it could work:

https://user-images.githubusercontent.com/75596775/226179378-42ce1e3f-23d1-4733-9655-3696fa124ba6.mp4

**Comment by peppy** [comment]
This can be done by implementing a `Container` type into the skinning system. That part is simple, but getting it to work well within the editor (ie. "drilling down" into multiple layers of containers) is going to be the harder part.

Personally I'm not quite ready to add this level of complexity to the skinning system.
